src/mapstorage/XmlMapStorage.cpp

```diff
@@ -27,11 +33,178 @@
 #include "progresscounter.h"
 #include "roomsaver.h"
 
+// ---------------------------- XmlMapStorage::Type ------------------------
+// list know enum types
+#define X_FOREACH_TYPE_ENUM(X) \
+    X(RoomAlignEnum) \
+    X(DoorFlagEnum) \
+    X(ExitFlagEnum) \
+    X(RoomLightEnum) \
+    X(RoomLoadFlagEnum) \
+    X(InfoMarkClassEnum) \
+    X(InfoMarkTypeEnum) \
+    X(RoomMobFlagEnum) \
+    X(RoomPortableEnum) \
+    X(RoomRidableEnum) \
+    X(RoomSundeathEnum) \
+    X(RoomTerrainEnum) \
+    X(Type)
+
+enum class XmlMapStorage::Type : uint {
+#define DECL(X) X,
+    X_FOREACH_TYPE_ENUM(DECL)
+#undef DECL
+};
+
+// ---------------------------- XmlMapStorage::Converter -----------------------
+class XmlMapStorage::Converter
+{
+public:
+    Converter();
+
+    // parse string containing an unsigned number.
+    // sets fail = true only in case of errors, otherwise fail is not modified
+    template<typename T>
+    static std::enable_if_t<std::is_integral<T>::value && std::is_unsigned<T>::value, T> //
```

Comment:
Do we even need the unsigned version of toNumber()?

Reply:
I have refactored this code to avoid QStringView, and the conversion from string to unsigned integer is now used to load room IDs.